util/cmd/release/main.go

```diff
@@ -109,9 +109,98 @@ func main() {
 	util.Execute(
 		"ghr",
 		"-t="+token,
+		"-n="+version,
+		"-b='"+notes(version)+"'",
 		"-u=googleapis",
 		"-r=gapic-generator-go",
 		"-c="+commitish,
 		version,
 		stagingDir)
 }
+
+// notes collects the commit messages since the previous tag, aggregates
+// dependency updates into one message, and returns formatted release notes.
+func notes(version string) string {
+	previous := util.ExecuteWithOutput(
+		"git",
+		"describe",
+		"--abbrev=0",
+		"--tags",
+		version+"^",
+	)
+	previous = strings.TrimSpace(previous)
+
+	output := util.ExecuteWithOutput(
+		"git",
+		"log",
+		previous+"..HEAD",
+		"--oneline",
+		"--pretty=format:%s",
+	)
+
+	var hasDeps bool
+	var gapic, bazel, gencli, chore, samples, other []string
+	for _, msg := range strings.Split(output, "\n") {
+		split := strings.Split(msg, ":")
+		comp := split[0]
+
+		switch comp {
+		case "gapic":
+			gapic = append(gapic, "*"+split[1])
+		case "bazel":
+			bazel = append(bazel, "*"+split[1])
+		case "gencli":
+			gencli = append(gencli, "*"+split[1])
+		case "chore(deps)":
+			hasDeps = true
+		case "chore":
+			chore = append(chore, "*"+split[1])
+		case "samples":
+			samples = append(samples, "*"+split[1])
+		default:
+			other = append(other, "* "+msg)
```

Comment:
Why a space after this `*` and not the others?

Reply:
Hm this might be a brittle assumption. But it's based on the assumption that one of the known components will be formatted with a space following the colon `gapic: fix some stuff`. That space shows up in `split[1]` as the first char. 

However, with unknown components or just malformed messages that are put in "other", the entire commit message is used (it's not split on the colon), so a space is added to separate the full `msg` from the `*`. 

Do you think I should strip white space on the `split` components and manually add the space? For example `samples = append(samples, "* " + strings.TrimSpace(split[1]))`.

Reply:
I think so. Minor change, but it will help in the accidental case.

Reply:
Done! Did it with a little less repeated code though :)